lib/ansible/modules/network/cloudengine/ce_snmp_traps.py

```diff
@@ -0,0 +1,552 @@
+      provider: "{{ cli }}"
+'''
+
+RETURN = '''
+changed:
+    description: check to see if a change was made on the device
+    returned: always
+    type: boolean
+    sample: true
+proposed:
+    description: k/v pairs of parameters passed into module
+    returned: always
+    type: dict
+    sample: {"feature_name": "all",
+             "state": "present"}
+existing:
+    description:
+        - k/v pairs of existing aaa server
+    type: dict
+    sample: {"snmp-agent trap": [],
+             "undo snmp-agent trap": []}
+end_state:
+    description: k/v pairs of aaa params after module execution
+    returned: always
+    type: dict
+    sample: {"snmp-agent trap": ["enable"],
+             "undo snmp-agent trap": []}
+updates:
+    description: command sent to the device
+    returned: always
+    type: list
+    sample: ["snmp-agent trap enable"]
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ce import get_config, load_config, ce_argument_spec, run_commands
+
+
+class SnmpTraps(object):
+    """ Manages SNMP trap configuration """
+
+    def __init__(self, **kwargs):
+        """ Class init """
+
+        # module
+        argument_spec = kwargs["argument_spec"]
+        self.spec = argument_spec
+        self.module = AnsibleModule(argument_spec=self.spec, supports_check_mode=True)
+
+        # config
+        self.cur_cfg = dict()
+        self.cur_cfg["snmp-agent trap"] = []
+        self.cur_cfg["undo snmp-agent trap"] = []
+
+        # module args
+        self.state = self.module.params['state']
+        self.feature_name = self.module.params['feature_name']
+        self.trap_name = self.module.params['trap_name']
+        self.interface_type = self.module.params['interface_type']
+        self.interface_number = self.module.params['interface_number']
+        self.port_number = self.module.params['port_number']
+
+        # state
+        self.changed = False
+        self.updates_cmd = list()
+        self.results = dict()
+        self.proposed = dict()
+        self.existing = dict()
+        self.existing["snmp-agent trap"] = []
+        self.existing["undo snmp-agent trap"] = []
+        self.end_state = dict()
+        self.end_state["snmp-agent trap"] = []
+        self.end_state["undo snmp-agent trap"] = []
+
+        commands = list()
+        cmd1 = 'display interface brief'
+        commands.append(cmd1)
+        self.interface = run_commands(self.module, commands)
+
+    def check_args(self):
+        """ Check invalid args """
+
+        if self.port_number:
+            if self.port_number.isdigit():
+                if int(self.port_number) < 1025 or int(self.port_number) > 65535:
+                    self.module.fail_json(
+                        msg='Error: The value of port_number is out of [1025 - 65535].')
+            else:
+                self.module.fail_json(
+                    msg='Error: The port_number is not digit.')
+
+        if self.interface_type:
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
```